packages/libs/eda/src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -1411,7 +1411,7 @@ export function histogramResponseToData(
   overlayVariable?: Variable,
   facetVariable?: Variable
 ): HistogramDataWithCoverageStatistics {
-  if (response.histogram.data.length === 0)
+  if (!response.histogram.data.find((item) => item.value.length > 0))
     throw Error(`Expected one or more data series, but got zero`);
```

Comment:
This is an old error message that was never/rarely seen by users mainly because null values were causing `io-ts` to complain first.

I have removed this test entirely and found that nothing bad happens. However there is an empty plot with no explanation to the user. Which is bad. However, the `PluginError` component should handle this for us, but `outputSize` being passed to it is `undefined`.  We could either ensure that `outputSize` is zero before it is passed, or change the login in `PluginError` to this:

```diff
diff --git a/packages/libs/eda/src/lib/core/components/visualizations/PluginError.tsx b/packages/libs/eda/src/lib/core/components/visualizations/PluginError.tsx
index 51956105a3..6f8a1ea0b8 100644
--- a/packages/libs/eda/src/lib/core/components/visualizations/PluginError.tsx
+++ b/packages/libs/eda/src/lib/core/components/visualizations/PluginError.tsx
@@ -23,7 +23,7 @@ export default function PluginError({ error, customCases, outputSize }: Props) {
   const fallbackErrorMessage =
     error == null ? '' : error instanceof Error ? error.message : String(error);
 
-  const emptyCase = () => (outputSize === 0 ? emptyCaseMessage : undefined);
+  const emptyCase = () => (!outputSize ? emptyCaseMessage : undefined);
 
   const errorContent =
     (customCases ?? [])
```

I think I prefer the latter.

Reply:
I realized that the latter did not work. It is because SAM's histogram always return outputSize = undefined. Thus, if we want to use PluginError, then we need to add an inline condition before passing to PluginError